### PR TITLE
Activate Local Snapshots

### PR DESCRIPTION
**Pull request by hmoog**
# Description

This PR replaces the refactored classes and activates local snapshots. It is not the last PR, yet since we want to integrate the missing repair routine and the requester thread but local snapshots should be fully functional already.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# Checklist:

_Please delete items that are not relevant._

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/com/iota/iri/service/ledger/impl/LedgerServiceImpl.java [hunk 1]
```diff
@@ -17,49 +17,58 @@
...
 
     /**
      * Holds a reference to the service instance containing the business logic of the milestone package.<br />
      */
-    private final MilestoneService milestoneService;
+    private MilestoneService milestoneService;
 
     /**
-     * Creates a service instance that allows us to perform ledger state specific operations.<br />
+     * This method initializes the instance and registers its dependencies.<br />
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/service/ledger/impl/LedgerServiceImpl.java]:
```suggestion
     * Initializes the instance and registers its dependencies.<br />
```

File: src/main/java/com/iota/iri/Iota.java [hunk 2]
```diff
@@ -73,10 +83,27 @@
...
+
+    public final LatestSolidMilestoneTrackerImpl latestSolidMilestoneTracker;
+
+    public final SeenMilestonesRetrieverImpl seenMilestonesRetriever;
+
+    public final LedgerServiceImpl ledgerService = new LedgerServiceImpl();
+
+    public final AsyncTransactionPruner transactionPruner;
+
+    public final MilestoneSolidifierImpl milestoneSolidifier;
```
[the remark this answers is not in the document]
Comment (hmoog) [reply src/main/java/com/iota/iri/Iota.java]: This is so we can use all the implementation specific methods like "init", "start" and so on without having to typecast them to their explicit type again. I could have the field be generic and then do

((MilestoneSolidifierImpl) milestoneSolidifier).init(...) but that kind of makes it more complicated imho

File: src/main/java/com/iota/iri/service/ledger/impl/LedgerServiceImpl.java [hunk 3]
```diff
@@ -17,49 +17,58 @@
...
+ * Creates a service instance that allows us to perform ledger state specific operations.<br />
  * <br />
  * This class is stateless and does not hold any domain specific models.<br />
  */
 public class LedgerServiceImpl implements LedgerService {
     /**
      * Holds the tangle object which acts as a database interface.<br />
      */
-    private final Tangle tangle;
+    private Tangle tangle;
```
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/service/ledger/impl/LedgerServiceImpl.java]: oh ofcourse

File: src/main/java/com/iota/iri/Iota.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/Iota.java]: hmm
Once we will do proper DI we will fix it

For now it is fine

**Pull request merged**